tools/run.py

```diff
@@ -0,0 +1,392 @@
+    nth_mixture['rirs'].append(riridx)
+    return nth_mixture
+
+
+def populate_mixture(cfg, sample_counter, fold_event_durs, fold_event_filenames, n_traj, traj_doas, max_polyphony, nth_mixture):
+    event_counter = 0
+
+    # populate each "layer" in the mixture
+    for layer in range(max_polyphony):
+
+        event_idx_in_layer, event_counter, sample_counter = get_layer_event_indices(fold_event_durs, cfg.MIXTURE_DUR - cfg.MIXTURE_LAYER_SILENCE, event_counter, sample_counter)
+        nevents_in_layer = len(event_idx_in_layer)
+
+        silence_gaps = get_silence_gaps(nevents_in_layer, cfg.MIXTURE_LAYER_SILENCE, cfg.MIN_GAP_BTW_LAYER_EVENTS)
+
+        # distribute each event in the timeline
+        time_idx = 0
+        for nl in range(nevents_in_layer):
+
+            event_duration_nl, riridx, sample_onoffsets, ev_traj, is_moving, is_flipped_moving, ev_speed, event_nl, time_idx = get_event_riridx_onoffset(nl, silence_gaps, event_idx_in_layer, fold_event_durs, fold_event_filenames, cfg, time_idx, n_traj, traj_doas)
+
+            nth_mixture = update_nth_mixture(nth_mixture, riridx, time_idx, event_duration_nl, sample_onoffsets, traj_doas, nl, layer, ev_traj, fold_event_filenames, is_moving, is_flipped_moving, ev_speed, event_nl, cfg)
+
+            time_idx += event_duration_nl
+
+    sort_idx = np.argsort(nth_mixture['event_onoffsets'][:,0])
+    nth_mixture['files'] = nth_mixture['files'][sort_idx]
+    nth_mixture['class'] = nth_mixture['class'][sort_idx]
+    nth_mixture['event_onoffsets'] = nth_mixture['event_onoffsets'][sort_idx]
+    #nth_mixture['sample_onoffsets'] = nth_mixture['sample_onoffsets'][sort_idx]
+    nth_mixture['trajectory'] = nth_mixture['trajectory'][sort_idx]
+    nth_mixture['isMoving'] = nth_mixture['isMoving'][sort_idx]
+    nth_mixture['isFlippedMoving'] = nth_mixture['isFlippedMoving'][sort_idx]
+    nth_mixture['speed'] = nth_mixture['speed'][sort_idx]
+    nth_mixture['rirs'] = np.array(nth_mixture['rirs'],dtype=object)
+    nth_mixture['rirs'] = nth_mixture['rirs'][sort_idx]
+    new_doas = np.zeros(len(sort_idx),dtype=object)
+    new_sample_onoffsets = np.zeros(len(sort_idx),dtype=object)
+    upd_idx = 0
+    for idx in sort_idx:
+        new_doas[upd_idx] = nth_mixture['doa_azel'][idx].T
+        new_sample_onoffsets[upd_idx] = nth_mixture['sample_onoffsets'][idx]
+        upd_idx += 1
+    nth_mixture['doa_azel'] = new_doas
+    nth_mixture['sample_onoffsets'] = new_sample_onoffsets 
+
+    return nth_mixture
+
+def main():
+    '''
+    main function to trigger the data generation
+    '''
+
+    # parse config arguments
+    args = parse_args()
+    cfg = load_config(args, args.path_to_config)
+    print('generating data using parameters:')
+    print(yaml.dump(dict(cfg), allow_unicode=True, default_flow_style=False))
+
+    
+    event_filenames = get_sound_event_filenames(cfg.PATH_TO_SOUND_EVENT_FILES)
+    event_durs = get_sound_event_filedurs(cfg.PATH_TO_SOUND_EVENT_DURS)
```

Comment:
right now I'm assuming that there will be a file containing the potential sound event filenames. This is not ideal, but it is the result of using `db_config_fsd._samplelist[0]['audiofile']` and `db_config_fsd._samplelist[0]['duration']`.

I do like the `scaper` idea of having a file structure:
```
- train
    - siren
        - siren1.wav
        - siren2.wav
        - some_siren_sound.wav
    - car_honk
        - honk.wav
        - beep.wav
    - human_voice
        - hello_world.wav
        - shouting.wav
- test
    - siren
        - siren3.wav
        - siren4.wav
        - some_siren_sound2.wav
    - car_honk
        - honk1.wav
        - beep1.wav
    - human_voice
        - hello_world1.wav
        - shouting1.wav
``` 

However, that still poses the challenge of retrieving the file duration in order to generate the metadata. Knowing the file duration ahead of time seems to be important for moving sound sources that could otherwise overshoot a given trajectories depending on how fast or slow it is.

Reply:
To circumvent this we could use `librosa.get_duration`, but we need to benchmark how fast or slow it may be.

Reply:
@adrianSRoman could you work on creating a script to convert the `db_config_fsd50k` into this file structure? From zero, to downloading the data, to putting it in place, and then replace these lines I have highlighted?

Reply:
Is the following the generic directory structure we want to adopt for any dataset?:
```
- <split1>
      - <class>
         - <track_id>
...
```
I do agree we should adopt a generic directory structure. Then, we can autogenerate any yaml files based on such structure, which will be more scalable in the long run.

Reply:
I agree with this even database restructuring; it's been quite a pain working with FSD50k for this reason. If anyone wants to claim this issue, it's hanging on #5 .

Reply:
Sounds like a plan. I will bring up a draft.